DOCS-12113-tracing-module-kt (#1635)
mulesoft/docs-connectors: tracing/1.0/modules/ROOT/pages/tracing-module-reference.adoc

> = Tracing Module 1.0 Reference

Reviewer: This is typically omitted from Connector editorial reviews: still autogenerated?

> Tracing module enables you to enhance your logs by adding, removing, and clearing variables from the logging context for a given Mule event. It also enables you to modify the correlation ID during the execution of a flow.

Reviewer: ditto re. order of operations

Default configuration

==== Parameters
[cols=".^20%,.^20%,.^35%,.^20%,^.^5%", options="header"]
|===
| Name | Type | Description | Default Value | Required
|Name | String | The name for this configuration. Connectors reference the configuration with this name. | | x
|===

== Operations

[[clear-logging]]
=== Clear Logging Variables
`<clear-logging-variables>`

Clears all the logging variables from the event logging context. This operation does not remove the correlation ID nor the current event processor because those variables are managed by Mule runtime engine.

==== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required
| Name | String | Display name for the operation  | |
|===

[[remove-logging]]
=== Remove Logging Variables
`<remove-logging-variable>`

Removes a variable from the event logging context.

==== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required
| Name | String | Display name for the operation. | |
| Variable name | String | Name of the variable to remove.  | x |
|===

[[set-logging]]
=== Set Logging Variables
`<set-logging-variable>`

> Sets a new logging variable and its value. The new variable will be present in the next operation execution for the event.

Reviewer: ```suggestion
Sets a new logging variable and its value. The new variable is present in the next operation execution for the event.
```
